### PR TITLE
sql-35: as-user-i-want-to-have-projects-page

### PR DESCRIPTION
**Pull request by IlliaLop**

### CODE REVIEW

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+{
+    public class ProjectService : BaseService, IProjectService
+    {
+        public ProjectService(SquirrelCoreContext context, IMapper mapper): base(context, mapper){ }
+
+        public async Task<ProjectDto> AddProjectAsync(ProjectDto projectDto)
+        {
+            var projectEntity = _mapper.Map<Project>(projectDto);
+            
+            projectEntity.DbEngine = projectDto.DbEngine;
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs]: Why you can't make it in mapper?

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            if (project is null)
+            {
+                throw new EntityNotFoundException();
+            }
+                 
+            _context.Projects.Remove(project);
+            
+            await _context.SaveChangesAsync();
+            
+            return _mapper.Map<ProjectDto>(project);
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs]: Why you return project if you just removed it?

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        public async Task<ProjectDto> GetProjectAsync(int projectId)
+        {
+            var project = await _context.Projects.FindAsync(projectId);
+            if (project is null)
+            {
+                throw new EntityNotFoundException();
+            }            
+            var projectDto = _mapper.Map<ProjectDto>(project);
+            projectDto.DbEngine = project.DbEngine;
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs]: The same

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            return projectDto;
+        }
+
+        public async Task<List<ProjectDto>> GetAllProjectsAsync()
+        {
+            var projects = await _context.Projects.ToListAsync();
+            var projectDtos = _mapper.Map<List<ProjectDto>>(projects);
+
+            return projectDtos;
+        }
```
Comment (Anatolii-Ivanov) [inline backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs]: return _mapper.Map...

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs (hunk 1 above)
Comment (IlliaLop) [reply backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs:20]: > Why you can't make it in mapper?
It was an accident, I'll clean it up now

Comment (IlliaLop) [reply backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs]: > Why you can't make it in mapper?

It was an accident, I'll clean it up now

File: backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs (hunk 2 above)
Comment (IlliaLop) [reply backend/Squirrel.Core/Squirrel.Core.BLL/Services/ProjectService.cs]: > Why you return project if you just removed it?

I return the status with a code and tell you which project was deleted. Should I remove it?

File: frontend/src/app/modules/projects/projects-page/projects-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            () => {
+                this.notificationService.error('Failed to load projects');
+            },
+        );
+    }
+
+    openCreateModal(): void {
+        const dialogRef = this.dialog.open(CreateProjectModalComponent, {
+            width: '35%',
+            height: '45%',
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/projects/projects-page/projects-page.component.ts]: Width in % can be really bad idea if you open on big or really small screen
Partially it's true for height

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by serhiiyasenev — APPROVED** [review]

**Pull request merged**